### PR TITLE
Update native_functions.yaml to add DefaultBackend.

### PR DESCRIPTION
**Pull request by ailzhang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46127 Try remove catchAllKenrel_.
* #45939 Move catchAll to Math
* #46126 Add guideline about which dispatch keyword to use in native_functions.yaml.
* **#45938 Update native_functions.yaml to add DefaultBackend.**

Differential Revision: [D24165889](https://our.internmc.facebook.com/intern/diff/D24165889)

### CODE REVIEW

**Comment by ailzhang** [comment]
The `native_functions.yaml` changes is generated from @ljk53 's static analysis https://github.com/pytorch/pytorch/pull/44963 (huge thanks!) plus a few added for ops in `VariableTypeManual.cpp`.

**Comment by ailzhang** [comment]
@bhosmer @ezyang May I get a review for this PR when you have time? ;) 
It's syncing derivatives.yaml's information to native_functions.yaml so there's a race condition between landing it and other PRs touching derivatives.yaml. Thanks!

**Comment by ezyang** [comment]
Don't forget to write the PR desc ;) ;) ;)

File: tools/codegen/gen.py [hunk 1]
```diff
@@ -897,7 +897,7 @@ def compute_registration_declarations(f: NativeFunction) -> str:
     comment_data : Dict[str, str] = {
         'schema': f'aten::{f.func}',
         'dispatch': str(f.dispatch is not None),
-        'math': str(f.dispatch is not None and 'Math' in f.dispatch)
+        'default': str(f.dispatch is not None and ('Math' in f.dispatch or 'DefaultBackend' in f.dispatch))
```
Comment (ezyang) [inline tools/codegen/gen.py:900]: Is there a paired XLA change? Or was math not being used yet.

Comment (ailzhang) [reply tools/codegen/gen.py:900]: Yea there's a paired XLA change

File: tools/autograd/gen_variable_type.py [hunk 2]
```diff
@@ -746,6 +746,13 @@ def gen_variable_type_shard(out, aten_declarations, template_path, suffix, heade
 
         # See Note [Manual catchAll kernels]
         assert (declaration['name'] in MANUAL_CATCHALL) == declaration['manual_kernel_registration']
+        # If you want to register a kernel to Autograd, you must make the op abstract.
+        # In other words, this op must have dispatch section in native_functions.yaml.
+        if declaration['name'] in MANUAL_AUTOGRAD_AND_TRACER or declaration['derivative']:
+            msg = "Did you add a formula for {}(or its functional variant) in derivatives.yaml? \
+                   If so please add a dispatch section for it with DefaultBackend in \
+                   native_functions.yaml.".format(declaration['name'])
```
Comment (ezyang) [inline tools/autograd/gen_variable_type.py:754]: nit: we got f-strings, use 'em!!

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
Don't bother addressing the review comments

File: tools/codegen/gen.py (hunk 1 above)
Comment (bhosmer) [reply tools/codegen/gen.py:900]: This is re-triggering my sense that `['Math', 'DefaultBackend']` is a semantically significant grouping and that we should define it somewhere (maybe not model.py, but somewhere), per the review comment in #45719. I don't know if it's urgent to do before landing this PR, but I'd like us to understand what common property it is that causes these two keys to show show up together, and ideally capture it in a definition.

**Review by bhosmer — APPROVED** [review]

**Pull request closed without merging**